integration-tests/src/test/java/org/apache/druid/tests/query/ITBroadcastJoinQueryTest.java

```diff
@@ -127,9 +130,22 @@ public void testBroadcastJoin() throws Exception
           replaceJoinTemplate(getResourceAsString(BROADCAST_JOIN_QUERIES_RESOURCE), BROADCAST_JOIN_DATASOURCE)
       );
     }
+
     finally {
       closer.close();
 
+      coordinator.unloadSegmentsForDataSource(BROADCAST_JOIN_DATASOURCE);
```

Comment:
As you mention in the description, these changes are probably not needed.
The closer.close() would have already internally called `coordinator.unloadSegmentsForDataSource()` and kept retrying until it succeeds at `AbstractIndexerTest.unloadAndKillData()`.

I think these changes would just add an extra retry.

Reply:
ah yeah, i forgot to look closer at what unloader was doing, i just added these changes initially when i was thrashing around trying to find why stuff was failing and on the surface didn't seem to be quite the same, but see now that this is repeating a subset of what unloader was doing, so have removed.